sdk/search/azure-search-documents/samples/async_samples/sample_vector_search_async.py

```diff
@@ -0,0 +1,196 @@
+        ),
+    ]
+    vector_search = VectorSearch(
+        algorithm_configurations=[VectorSearchAlgorithmConfiguration(name="my-vector-config", kind="hnsw")]
+    )
+    return SearchIndex(name=name, fields=fields, vector_search=vector_search)
+
+
+def get_hotel_documents():
+    docs = [
+        {
+            "hotelId": "1",
+            "hotelName": "Fancy Stay",
+            "description": "Best hotel in town if you like luxury hotels.",
+            "DescriptionVector": get_embeddings("Best hotel in town if you like luxury hotels."),
+            "category": "Luxury",
+        },
+        {
+            "HotelId": "2",
+            "HotelName": "Roach Motel",
+            "Description": "Cheapest hotel in town. Infact, a motel.",
+            "DescriptionVector": get_embeddings("Cheapest hotel in town. Infact, a motel."),
+            "Category": "Budget",
+        },
+        {
+            "HotelId": "3",
+            "HotelName": "EconoStay",
+            "Description": "Very popular hotel in town.",
+            "DescriptionVector": get_embeddings("Very popular hotel in town."),
+            "Category": "Budget",
+        },
+        {
+            "HotelId": "4",
+            "HotelName": "Modern Stay",
+            "Description": "Modern architecture, very polite staff and very clean. Also very affordable.",
+            "DescriptionVector": get_embeddings(
+                "Modern architecture, very polite staff and very clean. Also very affordable."
+            ),
+            "Category": "Luxury",
+        },
+        {
+            "HotelId": "5",
+            "HotelName": "Secret Point",
+            "Description": "One of the best hotel in town. The hotel is ideally located on the main commercial artery of the city in the heart of New York.",
+            "DescriptionVector": get_embeddings(
+                "One of the best hotel in town. The hotel is ideally located on the main commercial artery of the city in the heart of New York."
+            ),
+            "Category": "Boutique",
+        },
+    ]
+    return docs
+
+
+async def single_vector_search():
+    # [START single_vector_search]
+    query = "Top hotels in town"
+
+    search_client = SearchClient(service_endpoint, index_name, AzureKeyCredential(key))
+
+    async with search_client:
+        results = await search_client.search(
+            search_text="",
+            vector=Vector(value=get_embeddings(query), k=3, fields="descriptionVector"),
+            select=["hotelId", "hotelName"],
+        )
+
+        async for result in results:
+            print(result)
+    # [END single_vector_search]
+
+
+async def single_vector_search_with_filter():
+    # [START single_vector_search_with_filter]
+    query = "Top hotels in town"
+
+    search_client = SearchClient(service_endpoint, index_name, AzureKeyCredential(key))
+
+    async with search_client:
+        results = await search_client.search(
+            search_text="",
```

Comment:
does `search_text` need to be passed in if Im only doing a vector search ?

Reply:
Currently it is required.